File: easy/3637_trionic_array_i/solution.py
from typing import List


class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        numsLen = len(nums)

        p = 0
        while p < numsLen - 1 and nums[p] < nums[p + 1]:
            p += 1

        if p == 0 or p == numsLen - 1:
            return False

        q = p
        while q < numsLen - 1 and nums[q] > nums[q + 1]:
            q += 1

        if q == numsLen - 1:
            return False

        while q < numsLen - 1:
            if nums[q] >= nums[q + 1]:
                return False

            q += 1

        return q == numsLen - 1

File: easy/3637_trionic_array_i/Solution.java
class Solution {
  public boolean isTrionic(int[] nums) {
    final int numsLen = nums.length;
    int i = 0;

    for (; i < numsLen - 1 && nums[i] < nums[i + 1]; ++i) {}
    if (i == 0 || i == numsLen - 1) {
      return false;
    }

    for (; i < numsLen - 1 && nums[i] > nums[i + 1]; ++i) {}
    if (i == numsLen - 1) {
      return false;
    }

    for (; i < numsLen - 1 && nums[i] < nums[i + 1]; ++i) {}
    return i == numsLen - 1;
  }
}
